homework1/app.js

```diff
@@ -0,0 +1,297 @@
+        this.onClick = () => this.addItemToCart();
+        this.init();
+    }
+
+    init () {
+        this.root.addEventListener('click', this.onClick);
+    }
+
+    destroy () {
+    }
+
+    addItemToCart () {
+        // hint: you can use `dataset` to access data attributes
+        // See passing data from HTML to JavaScript from course note
+        let cartItems = this.store.cartItems || [];
+        // TODO: replace with actual item
+        console.log(this.root.dataset);
+        cartItems.push({
+            name: this.root.dataset.name,
+            price: this.root.dataset.price,
+            image: this .root.dataset.image
+        });
+        console.log(cartItems);
+        this.store.cartItems = cartItems;
+    }
+}
+
+class StatusTable {
+    // take dom element into JavaScript class to attachment events
+    constructor(root, store) {
+        this.root = root;
+        this.store = store;
+        init();
+    }
+
+    init () {
+        // attach click event listener to table header row on each column
+        render();
+    }
+
+    destroy () {
+        // remove all the events attached from init
+    }
+
+    sort (columnName) {
+        // after sorting the array of statuses, re render item to update view
+        render();
+    }
+
+    // render rows of items under table using root.innerHTML
+    render () {
+
+    }
+}
+
+// DOMContentLoaded event will allow us to run the following function when
+// everything is ready. Think of the following code will only be executed by
+// the end of document
+document.addEventListener('DOMContentLoaded', () => {
+	console.log('This is the starting point');
+    // use querySelector to find the table element (preferably by id selector)
+    // let statusTable = document.querySelector('');
+    // // use querySelector to find the cart element (preferably by id selector)
+    let cart = document.querySelector('.cart-table');
+    let checkoutButtons = document.querySelectorAll('.checkout-button');
+
+    console.log(cart);
+    console.log(checkoutButtons);
+
+    let store = new Store(window.localStorage);
+    // if (table) {
+    //     new StatusTable(table, store);
+    // }
+    if (cart) {
+        new Cart(cart, store);
+    }
+    if (checkoutButtons && checkoutButtons.length) {
+        for (var i = 0; i < checkoutButtons.length; i ++) {
+            new CheckoutButton(checkoutButtons[i], store);
+        }
+    }
+});
+class Inventory {
+    constructor(root, store) {
+        this.root = root;
+        this.store = store;
+        this.init();
+    }
+
+    init () {
+        this.render();
+        // TODO: attach event listeners like click to remove items after rendering
+    }
+
+    destroy () {
+        // TODO: remove event listeners added from the init above
+    }
+
+    removeItem (itemName) {
+        console.log(this.root.dataset);
+        items.remove({
+            name:this.root.dataset.name,
+            price:this.root.dataset.price
+        });
```

Comment:
this method will simply throw null pointer exception on "items". It's never defined anywhere in the function scope.